deploy/checkProxy.py

```diff
@@ -0,0 +1,162 @@
+    for opt, arg in opts:
+        if opt == "--help":
+            print(main.__doc__)
+            sys.exit(0)
+        if opt == "--proxy":
+            proxy = arg
+            if proxy.startswith("~/"):
+                proxy = os.getenv('HOME') + proxy[1:]
+            if not os.path.exists(proxy):
+                print("Proxy File does not exist")
+                sys.exit(2)
+        if opt == "--mail":
+            mail = arg
+        if opt == "--myproxy":
+            myproxy = arg
+        if opt == "--send-mail":
+            sendMail = arg
+        if opt == "--time":
+            time = int(arg)
+            if time < 1:
+                print("Invalid time format. Check the options: {}".format(main.__doc__))
+                raise sys.exit(3)
+        if opt == "--verbose":
+            verbose = True
+
+    command = ["voms-proxy-info", "-file", str(proxy)]
+    p = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    out, _err = p.communicate()
+    proxyInfo = [line for line in out.split('\n') if line]
+    processTimeLeft(host, sendMail, verbose, proxyInfo, time, mail)
+
+    if myproxy:
+        os.environ["X509_USER_CERT"] = proxy
+        os.environ["X509_USER_KEY"] = proxy
+        command = ["myproxy-info", "-v", "-l", "amaltaro", "-s", "myproxy.cern.ch", "-k", "amaltaroCERN"]
+        p = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        out, err = p.communicate()
+        proxyInfo = [line for line in out.split('\n') if line]
+        processTimeLeft(host, sendMail, verbose, proxyInfo, time, mail)
+
+
+def sendMailNotification(mail, message, proxyInfo='', verbose=False):
+    host = os.getenv('HOSTNAME')
+    if verbose:
+        print("Host: {}".format(host))
+    # append proxy info to message
+    for line in proxyInfo:
+        message += "%s\n" % line
+    # echo %msg | mail -s 'HOST proxy status' MAIL_ADD
+    command = " echo \"%s\" | " % (message)
+    command += "mail -s '%s: Proxy status'" % (host)
+    command += " %s" % (mail)
+
+    if verbose:
+        print("Running email command: {}".format(command))
+    retCode = os.system(command)
+    if verbose:
+        print("Email notification exit code: {}".format(retCode))
+
+
+def processTimeLeft(host, sendMail, verbose, proxyInfo, time, mail):
+    """
+    Receive the whole proxy info and return its time left.
+    In case no proxy information is provided, it sends an
+    email warning the user.
+    """
+    if proxyInfo:
+        if verbose:
+            print('Proxy information: {}'.format(proxyInfo))
+        timeLeft = []
+        for line in proxyInfo:
+            if line.find('timeleft') > -1:
+                dateReg = re.compile('\d{1,3}[:/]\d{2}[:/]\d{2}')
+                timeLeft = dateReg.findall(line)[0]
+                timeLeft = timeLeft.split(':')[0]
+                continue
+    else:
+        msg = "Valid proxy file in %s not found. " % host
```

Comment:
I would change this message to:
"No valid proxy was found in %s"

Reply:
Changing